apps/admin/backend/src/reports/helpers.ts

```diff
@@ -0,0 +1,22 @@
+import { tmpNameSync } from 'tmp';
+import { promisify } from 'util';
+import { emptyDir as emptyDirCallback } from 'fs-extra';
+import { Workspace } from '../util/workspace';
+
+const emptyDir = promisify(emptyDirCallback);
+
+/**
+ * Returns the path of a temporary file in the workspace's preview directory.
+ * A PDF file should be written to the path, which will be served to the
+ * frontend through a static route.
+ *
+ * On each call, the previous preview file is deleted. This aims to prevent
+ * memory leaks and to make stale previews fail loudly.
```

Comment:
Technically this wouldn't be a memory leak, but a series of files on disk that never get cleaned up. I guess a more general term might be "resource leak", but that feels dissatisfying. Maybe "storage leak"?